src/vs/base/browser/ui/menu/menu.ts

```diff
@@ -796,13 +790,7 @@ class SubmenuMenuActionViewItem extends BaseMenuActionViewItem {
 			this.submenuDisposables.add(this.parentData.submenu.onDidCancel(() => {
 				this.parentData.parent.focus();
 
-				if (this.parentData.submenu) {
```

Comment:
@jeanp413 credit to you for calling this out in your PR. In this PR it became necessary to fix without further duplicating code.